x-pack/test/functional/services/machine_learning/job_wizard_common.ts

```diff
@@ -358,6 +388,39 @@ export function MachineLearningJobWizardCommonProvider({ getService }: FtrProvid
       await this.assertDateRangeSelection(expectedStartDate, expectedEndDate);
     },
 
+    async ensureAdditionalSettingsSectionOpen() {
+      await retry.tryForTime(5000, async () => {
+        if ((await testSubjects.exists(additionalSettingsSectionSelector())) === false) {
+          await testSubjects.click('mlJobWizardToggleAdditionalSettingsSection');
+          await testSubjects.existOrFail(additionalSettingsSectionSelector(), { timeout: 1000 });
+        }
+      });
+    },
+
+    async ensureNewCustomUrlFormModalOpen() {
+      await retry.tryForTime(5000, async () => {
+        if ((await testSubjects.exists(newCustomUrlFormModal())) === false) {
+          await testSubjects.click('mlJobOpenCustomUrlFormButton');
+          await testSubjects.existOrFail(newCustomUrlFormModal(), { timeout: 1000 });
+        }
+      });
+    },
+
+    async addCustomUrl() {
+      await this.ensureNewCustomUrlFormModalOpen();
+      // Fill-in the form
+      await customUrls.setCustomUrlLabel('check-kibana-dashboard');
+      // Add custom URL
+      await customUrls.addCustomUrl();
+
+      await customUrls.assertCustomUrlItem(0);
+    },
+
+    async assignCalendar(calendarId: string) {
```

Comment:
As mentioned above: I would move the `ensureAdditionalSettingsSectionOpen` to `addCalendar`, then this extra method is not needed any more.

Reply:
Removed in https://github.com/elastic/kibana/pull/52269/commits/2e174c680610465ccf6c4c2e4242034c2665b473